CMakeLists.txt

```diff
@@ -0,0 +1,79 @@
+# SPDX-License-Identifier: BSD-3-Clause
+
+cmake_minimum_required(VERSION 3.25)
```

Comment:
Is there any documentation that we need to add to the README on how to use this? I personally wouldn't know what `cmake` incantation to use to get this to work.

Reply:
Yeah, probably a good idea to document it.  The standard "BSG" applies here:

```cmd
cmake -B out -G Ninja -S .
```

Reply:
I think that I will do the documentation as a follow up.